llvm/include/llvm/CodeGen/MachineModuleInfo.h

```diff
@@ -124,10 +124,10 @@ class MachineModuleInfo {
   const LLVMTargetMachine &getTarget() const { return TM; }
 
   const MCContext &getContext() const {
-    return ExternalContext ? *ExternalContext : Context;
+    return ExternalContext ? *ExternalContext : *Context;
```

Comment:
Can we get rid of this ExternalContext business? I don't understand why it's here. If it's necessary to construct with an external context, I would expect that to be the only way to construct MMI

Reply:
Maybe? I don't understand why it's here either (and my PR is doing anything with it 😛). Any chance we can decide the fate of `ExternalContext` in a separate PR since it's out of the scope of this one?

Reply:
I'd say it's the same scope, for what should own the MCContext

Reply:
The question that needs to be answered is "who owns the MCContext". I think the answer should be definitely MachineModuleInfo, or definitively an external owner. Having 2 modes here is a bad API

Reply:
I agree; See my comments [below](https://github.com/llvm/llvm-project/pull/104834#issuecomment-2299180092). IMO it's better to have an external owner. It will bring MMI closer to how `llvm::Module` is managed.

Reply:
> Having 2 modes here is a bad API

I personally agree with this in principal, but again, I don't have much context on why these two are here originally, so I'd prefer not having to decide the fate of this API while not understanding the original intension of the design. If anyone else has more context, I'd really love to hear and happy to make another PR to improve this.